### PR TITLE
Feature/tb 41 42 43 form constructor

### PR DESCRIPTION
**Pull request by 6040H0K**

### CODE REVIEW

File: db/botSettings.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+const sequelize = require('./db.js').sequelize;
+const DataTypes = require('./db.js').DataTypes;
+// model of BotSettings
+const botSettings = sequelize.define('botSettings', {
```
Comment (Maksym-Lapshyn) [inline db/botSettings.js]: You should decide if you need to use plural or singular names in your sequelize models and stick to the single variant.

File: db/botSettings.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    allowNull: false,
+    primaryKey: true,
+},
+botToken:{
+    type: DataTypes.STRING,
+},
+emojies:{
+    type: DataTypes.JSON,
+}
+}, {})
```
Comment (Maksym-Lapshyn) [inline db/botSettings.js]: Do not forget to add empty lines before exports and return statements.

File: index.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+const pg = require('pg');
+const cors = require('cors');
+const routesBotSettings = require('./routes/botSettings.js');
+const routesLanguageProfile = require('./routes/languageProfile.js');
+
+const app = express();
+app.use(cors());
+app.use(express.json());
+
+app.listen(3000, () => {
```
Comment (Maksym-Lapshyn) [inline index.js:19]: I suggest to start the app in the very end of the file once you have your middleware set.

File: package.json [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  "license": "ISC",
+  "bugs": {
+    "url": "https://github.com/Glexen/matchbot-api/issues"
+  },
+  "homepage": "https://github.com/Glexen/matchbot-api#readme",
+  "dependencies": {
+    "cors": "^2.8.5",
+    "dotenv": "^16.0.3",
+    "express": "^4.18.2",
+    "nodemon": "^2.0.21",
```
Comment (Maksym-Lapshyn) [inline package.json:26]: Read about dependencies and dev dependencies in npm - https://docs.npmjs.com/specifying-dependencies-and-devdependencies-in-a-package-json-file

File: routes/botSettings.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
+const express = require('express')
+const botSettings = require('../db/botSettings.js')
+const router = express.Router()
+
+module.exports = router;
+// reset endpoint, use only in testing
+router.get('/start', async (req, res) => {
+    await botSettings.destroy({where: {}})
```
Comment (Maksym-Lapshyn) [inline routes/botSettings.js]: Make sure to use or not to use semicolons and stick to this one decision.

File: routes/botSettings.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    let bot = await botSettings.build({'id': 1,
```
Comment (Maksym-Lapshyn) [inline routes/botSettings.js]: Any reasons for using let instead of const here?

File: routes/botSettings.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+                    'botToken': 'Token',
```
Comment (Maksym-Lapshyn) [inline routes/botSettings.js]: Formatting is particularly odd here, fix it. Try using automatic formatting to achieve that. https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode

Also read about linters and add one to the project. Add the one that is most popular for express. https://eslint.org

File: routes/botSettings.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+                        {'like':'👍',
+                        'dislike':'👎',
+                        'message':'📝',
+                        'report':'‼️'}})
+    bot.save()
+    res.send(200)
+})
+// update data endpoint
+router.put('/change', async (req, res) => {
+    let bot = await botSettings.findByPk(1)
```
Comment (Maksym-Lapshyn) [inline routes/botSettings.js]: Is it "bot" or "settings"?

File: routes/botSettings.js [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+    bot.botToken = req.body.botToken;
+    bot.emojies = {'like':req.body.like,
+                    'dislike':req.body.dislike,
```
Comment (Maksym-Lapshyn) [inline routes/botSettings.js]: Never try to indent like that. Horizontal formatting is a relic of past.

**Review by Maksym-Lapshyn — CHANGES_REQUESTED** [review]

File: routes/botSettings.js [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+                    'emojies':
+                        {'like':'👍',
+                        'dislike':'👎',
+                        'message':'📝',
+                        'report':'‼️'}})
+    bot.save()
+    res.send(200)
+})
+// update data endpoint
+router.put('/change', async (req, res) => {
```
Comment (Maksym-Lapshyn) [inline routes/botSettings.js]: "update" is a better word than "change" for such tasks. If you only update bot token, then call this "update-token" to be more declarative.

File: routes/botSettings.js [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+    bot.emojies = {'like':req.body.like,
+                    'dislike':req.body.dislike,
+                    'message':req.body.message,
+                    'report':req.body.report};
+    await bot.save()
+    res.send(200)
+})
+// get data endpoint
+router.get('/get', async (req, res) => {
+    let bot = await botSettings.findByPk(1)
```
Comment (Maksym-Lapshyn) [inline routes/botSettings.js]: Why do you get by pk and hardcode it?

**Review by Maksym-Lapshyn — CHANGES_REQUESTED** [review]

File: routes/botSettings.js (hunk 11 above)
Comment (6040H0K) [reply routes/botSettings.js]: Because, its model like Singleton, and we have only one instance of model BotSetting, but maybe I must to use findOne?

Comment (Maksym-Lapshyn) [reply routes/botSettings.js]: Got it. That way I would recommend you to put it into the app's configuration. Read more here - https://www.freecodecamp.org/news/how-to-use-node-environment-variables-with-a-dotenv-file-for-node-js-and-npm/

File: package.json (hunk 4 above)
Comment (Maksym-Lapshyn) [reply package.json:26]: No action is needed atm.

File: routes/botSettings.js (hunk 5 above)
Comment (6040H0K) [reply routes/botSettings.js]: This file renamed to botSetting.js check it plz

File: routes/botSettings.js (hunk 6 above)
Comment (6040H0K) [reply routes/botSettings.js]: This file renamed to botSetting.js check it plz

File: routes/botSettings.js (hunk 7 above)
Comment (6040H0K) [reply routes/botSettings.js]: This file renamed to botSetting.js check it plz

File: routes/botSettings.js (hunk 8 above)
Comment (6040H0K) [reply routes/botSettings.js]: This file renamed to botSetting.js check it plz

File: routes/botSettings.js (hunk 9 above)
Comment (6040H0K) [reply routes/botSettings.js]: This file renamed to botSetting.js check it plz

File: routes/languageProfile.js [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+  language.save()
+  res.send(200)
+})
+
+router.get("/all", async (req, res) => {
+  let languages = await languageProfile.findAll()
+  res.send(JSON.stringify(languages, null, 2))
+})
+
+router.get("/one/:pk", async (req, res) => {
```
Comment (Maksym-Lapshyn) [inline routes/languageProfile.js]: Routes are still not REST-ful

File: routes/botSetting.js [hunk 13]
```diff
@@ -0,0 +1,68 @@
+const express = require("express")
+const botSetting = require("../db/botSetting.js")
+const router = express.Router()
+require('dotenv').config();
```
Comment (Maksym-Lapshyn) [inline routes/botSetting.js]: I still see that sometimes you use semicolons and sometimes do not. Update your eslint config to require them

**Review by Maksym-Lapshyn — CHANGES_REQUESTED** [review]

**Review by Maksym-Lapshyn — APPROVED** [review]

**Review by Maksym-Lapshyn — DISMISSED** [review]

**Review by artem-alie — DISMISSED** [review]

**Review by artem-alie — DISMISSED** [review]

**Review by artem-alie — APPROVED** [review]

**Pull request merged**